timeboost-types/src/bundle.rs

```diff
@@ -0,0 +1,463 @@
+        let mut this = Self {
+            chain,
+            epoch,
+            data,
+            kid,
+            hash: [0; 32],
+        };
+        this.update_hash();
+        this
+    }
+
+    fn update_hash(&mut self) {
+        let digest = self.commit();
+        self.hash = digest.into();
+    }
+}
+
+impl Bundle {
+    pub fn chain_id(&self) -> ChainId {
+        self.chain
+    }
+
+    pub fn epoch(&self) -> Epoch {
+        self.epoch
+    }
+
+    pub fn data(&self) -> &Bytes {
+        &self.data
+    }
+
+    pub fn kid(&self) -> Option<KeysetId> {
+        self.kid
+    }
+
+    pub fn digest(&self) -> &[u8; 32] {
+        &self.hash
+    }
+
+    pub fn set_data(&mut self, d: Bytes) {
+        self.data = d;
+        self.update_hash()
+    }
+
+    #[cfg(feature = "arbitrary")]
+    pub fn arbitrary(u: &mut arbitrary::Unstructured<'_>) -> arbitrary::Result<Bundle> {
+        use alloy_rlp::Encodable;
+        use arbitrary::Arbitrary;
+        let t: Transaction = loop {
+            let candidate: Transaction = arbitrary::Arbitrary::arbitrary(u)?;
+            if let TxEnvelope::Eip4844(ref eip4844) = candidate.0 {
+                if eip4844.tx().clone().try_into_4844_with_sidecar().is_ok() {
+                    // Avoid generating 4844 Tx with blobs of size 131 KB
+                    continue;
+                }
+            }
+            break candidate;
+        };
+
+        let mut d = Vec::new();
+        t.encode(&mut d);
+        let c = ChainId::default();
+        let e = Epoch::now() + bool::arbitrary(u)? as u64;
+        let encoded = ssz::ssz_encode(&vec![&d]);
+        let k = None;
+        let b = Bundle::new(c, e, encoded.into(), k);
+
+        Ok(b)
+    }
+}
+
+impl Committable for Bundle {
+    fn commit(&self) -> Commitment<Self> {
+        RawCommitmentBuilder::new("PriorityBundle")
+            .field("chain", self.chain_id().commit())
+            .field("epoch", self.epoch().commit())
+            .var_size_field("data", self.data())
+            .field("keysetid", self.kid().unwrap_or_default().commit())
+            .finalize()
+    }
+}
+
+pub trait State {}
+#[derive(Debug, Clone, PartialEq, Eq, Hash, PartialOrd, Ord, Serialize, Deserialize)]
+pub struct Signed {
+    signature: Signature,
+}
+pub struct Unsigned {}
+impl State for Signed {}
+impl State for Unsigned {}
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct PriorityBundle<S: State> {
+    bundle: Bundle,
+    auction: Address,
+    seqno: SeqNo,
+    hash: [u8; 32],
+    signed: S,
+}
+
+impl<S: State> PriorityBundle<S> {
+    pub fn bundle(&self) -> &Bundle {
+        &self.bundle
+    }
+
+    pub fn auction(&self) -> Address {
+        self.auction
+    }
+
+    pub fn seqno(&self) -> SeqNo {
+        self.seqno
+    }
+
+    // https://github.com/OffchainLabs/nitro/blob/1e16dc408d24a7784f19acd1e76a71daac528a22/timeboost/types.go#L206
+    pub fn as_bytes(&self) -> Vec<u8> {
```

Comment:
```suggestion
    pub fn to_bytes(&self) -> Vec<u8> {
```
https://rust-lang.github.io/api-guidelines/naming.html#ad-hoc-conversions-follow-as_-to_-into_-conventions-c-conv

Reply:
Thanks for the pointer. Fixed.